src/core/Akka/Actor/Props.cs

```diff
@@ -533,88 +500,152 @@ public Props WithDeploy(Deploy deploy)
         public Props WithSupervisorStrategy(SupervisorStrategy supervisorStrategy)
         {
-            Props copy = Copy();
+            var copy = Copy();
             copy.SupervisorStrategy = supervisorStrategy;
             return copy;
         }
 
         //TODO: use Linq Expressions so compile a creator
         //cache the creator
         /// <summary>
-        /// Creates a new actor using the configured actor producer.
-        /// 
-        /// <remarks>
-        /// This method is only useful when called during actor creation by the ActorSystem.
-        /// </remarks>
+        ///     Creates a new actor using the configured actor producer.
+        ///     <remarks>
+        ///         This method is only useful when called during actor creation by the ActorSystem.
+        ///     </remarks>
         /// </summary>
         /// <exception cref="TypeLoadException">
-        /// This exception is thrown if there was an error creating an actor of type <see cref="Props.Type"/>
-        /// with the arguments from <see cref="Props.Arguments"/>.
+        ///     This exception is thrown if there was an error creating an actor of type <see cref="Props.Type" />
+        ///     with the arguments from <see cref="Props.Arguments" />.
         /// </exception>
         /// <returns>The newly created actor</returns>
         public virtual ActorBase NewActor()
         {
             var type = Type;
             var arguments = Arguments;
-            try {
-                return producer.Produce();
-            } catch (Exception e) {
-                throw new TypeLoadException($"Error while creating actor instance of type {type} with {arguments.Length} args: ({StringFormat.SafeJoin(",", arguments)})", e);
+            try
+            {
+                return _producer.Produce();
+            }
+            catch (Exception e)
+            {
+                throw new TypeLoadException(
+                    $"Error while creating actor instance of type {type} with {arguments.Length} args: ({StringFormat.SafeJoin(",", arguments)})",
+                    e);
             }
         }
 
         /// <summary>
-        /// Creates a copy of the current instance.
+        ///     Creates a copy of the current instance.
         /// </summary>
-        /// <returns>The newly created <see cref="Akka.Actor.Props"/></returns>
+        /// <returns>The newly created <see cref="Akka.Actor.Props" /></returns>
         protected virtual Props Copy()
         {
-            return new Props(Deploy, inputType, Arguments) { SupervisorStrategy = SupervisorStrategy };
+            return new Props(Deploy, _inputType, Arguments) { SupervisorStrategy = SupervisorStrategy };
+        }
+
+        [Obsolete("we should not be calling this method. Pass in an explicit IIndirectActorProducer reference isntead.")]
+        private static IIndirectActorProducer CreateProducer(Type type, object[] args)
```

Comment:
Source of most of the weirdness I identified in https://github.com/akkadotnet/akka.net/issues/3599 stem from this method, which conflates accepting a `Type` for an actor with a `Type` for an `IIndirectActorProducer`, which has the effect of making it really difficult to reason about what is happening internally and performing reflection twice for all actor invocations.